lib/puppet/parser/functions/validate_present.rb

```diff
@@ -0,0 +1,15 @@
+module Puppet::Parser::Functions
```

Comment:
I think we can get rid of this since we have puppet data types. It means changing `Optional[String] $certs_tar` to `String[1] $certs_tar`. The fqdn check should already be redundant. It also means we can remove the `if $certs_tar` since that looks like a useless check.

Reply:
According to the docs it's letting us use `newfunction` https://docs.puppet.com/puppet/4.0/yard/Puppet/Parser/Functions.html#newfunction-class_method

Do you mean it's somehow redundant?

Reply:
I mean the whole function can be removed since we can simply replace all function calls with Puppet 4 data types.